### DISCUSSION TITLE
Enhancement: Add Internationalization (i18n) Support in UI

### DISCUSSION THREAD

**Discussion by iambocai** [PrefectHQ/prefect#12251] [Ideas]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

The current UI only supports English

### Describe the proposed behavior

The UI supports localized language display based on the user's browser language(`navigator.language`), and also supports users to switch the display language on the page.

### Example Use

![image](https://github.com/PrefectHQ/prefect/assets/2259715/619941ce-1358-4b0d-9102-20a891c3f0f3)

### Additional context

[prefect-ui-library](https://github.com/PrefectHQ/prefect-ui-library) seems have some i18n related files, but the [index.ts](https://github.com/PrefectHQ/prefect-ui-library/blob/main/src/localization/index.ts) seems not ready.

**Comment by xuzheliang135** [comment]
Is there any progress?‌
